### PR TITLE
Added Email button for the event list page

### PR DESCRIPTION
**Pull request by arpargo**
## Description :sparkles:
Send Email button to send notification information to event creator in the event listing page.

## Issues :bug:
The email information has to be parsed from the createdBy string which usually contains email address. There is potential that if the createdBy field is malformed that this feature does not work.

### Closes :no_good_woman:

**[LINK-1266](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1266):**

[LINK-1266]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/domain/app/i18n/fi.json [hunk 1]
```diff
@@ -538,7 +539,8 @@
         "warningCannotPublishSubEvent": "Sarjan alatapahtumia ei voi julkaista.",
         "warningDeletedEvent": "Poistettuja tapahtumia ei voi muokata.",
         "warningEventInPast": "Menneisyydessä olevia tapahtumia ei voi muokata.",
-        "warningNoRightsToEdit": "Sinulla ei ole oikeuksia muokata tätä tapahtumaa."
+        "warningNoRightsToEdit": "Sinulla ei ole oikeuksia muokata tätä tapahtumaa.",
+        "warningNoEmailFound": "Could not find email address linked to this event"
```
Comment (jorilindell) [inline src/domain/app/i18n/fi.json:543]: Finnish translation missing

File: src/domain/app/i18n/sv.json [hunk 2]
```diff
@@ -538,7 +539,8 @@
         "warningCannotPublishSubEvent": "Ett underevenemang i en serie kan inte publiceras.",
         "warningDeletedEvent": "Raderade evenemang kan inte redigeras.",
         "warningEventInPast": "Tidigare evenemnag kan inte redigeras.",
-        "warningNoRightsToEdit": "Du har inte tillräckliga rättigheter för att redigera detta evenemang."
+        "warningNoRightsToEdit": "Du har inte tillräckliga rättigheter för att redigera detta evenemang.",
+        "warningNoEmailFound": "Could not find email address linked to this event"
```
Comment (jorilindell) [inline src/domain/app/i18n/sv.json:543]: Swedish translation missing

File: src/domain/event/utils.ts [hunk 3]
```diff
@@ -1235,6 +1248,10 @@ const getUpdateEventActionWarning = (
     return t('event.form.editButtonPanel.warningEventInPast');
   }
 
+  if (noEmailFound && action == EVENT_ACTIONS.SEND_EMAIL) {
+    return t('event.form.editButtonPanel.warningNoEmailFound');
+  }
```
Comment (jorilindell) [inline src/domain/event/utils.ts:1253]: Could add a unit test to test that warning is shown if email is not found from created by attribure

File: src/domain/event/utils.ts [hunk 4]
```diff
@@ -1137,6 +1141,10 @@ export const getIsButtonVisible = ({
   }
 };
 
+const validateCreatedBy = (createdBy: string | null | undefined): boolean => {
+  return createdBy !== (null || undefined || '' || ' - ');
+};
```
Comment (jorilindell) [inline src/domain/event/utils.ts:1146]: Could add a unit test to test that created by is parsed correctly

File: src/domain/events/eventActionsDropdown/EventActionsDropdown.tsx [hunk 5]
```diff
@@ -64,6 +65,18 @@ const EventActionsDropdown: React.FC<EventActionsDropdownProps> = ({
...
+    let targetEmail = '';
+    if (event.createdBy != null && event.createdBy != undefined) {
+      targetEmail = event.createdBy.split('-')[1].trim();
+    }
+    window.location.href =
+      'mailto:' +
+      targetEmail +
+      '?subject=' +
+      getLocalisedString(event.name, locale);
+  };
```
Comment (jorilindell) [inline src/domain/events/eventActionsDropdown/EventActionsDropdown.tsx:78]: Could add a unit test to test that sendEmail + window.location.href functions are called with correct values.

**Review by jorilindell — CHANGES_REQUESTED** [review]
Code looks good. Added couple of comments about unit tests and missing translations. At least translations should be fixed.

File: src/domain/event/utils.ts [hunk 6]
```diff
@@ -1071,9 +1071,9 @@ export const checkCanUserDoAction = ({
   organizationAncestors: OrganizationFieldsFragment[];
   user?: UserFieldsFragment;
 }): boolean => {
-  const { isDraft, publisher } = event
+  const { isDraft, publisher, createdBy } = event
```
Comment (mikkojamG) [inline src/domain/event/utils.ts:1074]: It looks like this createBy constant is not actually used 🤔

File: src/domain/event/utils.ts [hunk 7]
```diff
@@ -1101,10 +1101,13 @@ export const checkCanUserDoAction = ({
     case EVENT_ACTIONS.ACCEPT_AND_PUBLISH:
     case EVENT_ACTIONS.PUBLISH:
     case EVENT_ACTIONS.UPDATE_PUBLIC:
+    case EVENT_ACTIONS.SEND_EMAIL:
       return isAdminUser;
     case EVENT_ACTIONS.UPDATE_DRAFT:
       return isRegularUser || isAdminUser;
   }
+
+  return false;
```
Comment (mikkojamG) [inline src/domain/event/utils.ts:1110]: This return appears to be unreachable, probably should be moved to default case of the switch statement 🤔

File: src/domain/event/utils.ts [hunk 8]
```diff
@@ -1137,6 +1141,10 @@ export const getIsButtonVisible = ({
   }
 };
 
+const validateCreatedBy = (createdBy: string | null | undefined): boolean => {
+  return createdBy !== (null || undefined || '' || ' - ');
```
Comment (mikkojamG) [inline src/domain/event/utils.ts:1145]: I think we would have the same effect with `return Boolean(createdBy) || createdBy !== ' - '` 🤔

File: src/domain/event/utils.ts [hunk 9]
```diff
@@ -1235,6 +1248,10 @@ const getUpdateEventActionWarning = (
     return t('event.form.editButtonPanel.warningEventInPast');
   }
 
+  if (noEmailFound && action == EVENT_ACTIONS.SEND_EMAIL) {
```
Comment (mikkojamG) [inline src/domain/event/utils.ts:1251]: eslint warns should have triple equaliser here, `action === EVENT_ACTIONS.SEND_EMAIL`

File: src/domain/events/eventActionsDropdown/EventActionsDropdown.tsx [hunk 10]
```diff
@@ -64,6 +65,18 @@ const EventActionsDropdown: React.FC<EventActionsDropdownProps> = ({
     navigate(`/${locale}${ROUTES.CREATE_EVENT}`);
   };
 
+  const sendEmail = () => {
+    let targetEmail = '';
+    if (event.createdBy != null && event.createdBy != undefined) {
```
Comment (mikkojamG) [inline src/domain/events/eventActionsDropdown/EventActionsDropdown.tsx:70]: I think `if (event.created) {...}` would give the same effect

File: src/domain/events/eventActionsDropdown/EventActionsDropdown.tsx [hunk 11]
```diff
@@ -64,6 +65,18 @@ const EventActionsDropdown: React.FC<EventActionsDropdownProps> = ({
...
+      targetEmail = event.createdBy.split('-')[1].trim();
+    }
+    window.location.href =
```
Comment (mikkojamG) [inline src/domain/events/eventActionsDropdown/EventActionsDropdown.tsx:73]: `mailto:${targetEmail}?subject=${getLocalisedString(event.name, locale)}` would also give the same effect 🤔

**Review by mikkojamG — COMMENTED** [review]
Well done, works very nicely 👌 Left couple of comments

**Review by mikkojamG — APPROVED** [review]
LGTM ✅

**Review by jorilindell — APPROVED** [review]
LGTM

**Pull request merged**